util/test/sub_test

```diff
@@ -504,8 +504,18 @@ def get_exec_log_name(execname, comp_opts_count=None, exec_opts_count=None):
 # Use testEnv to process skipif files, it works for executable and
 # non-executable versions
 def runSkipIf(skipifName):
-    skiptest = subprocess.Popen([utildir+'/test/testEnv', './'+skipifName], stdout=subprocess.PIPE).communicate()[0]
-    return skiptest
+    p = subprocess.Popen([utildir+'/test/testEnv', './'+skipifName], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
+    (stdout, stderr) = p.communicate()
+    status = p.returncode
+    stderr = stderr.strip()
+    if status:
+        if stderr:
+            sys.stdout.write(stderr+'\n')
+        Fatal('%s: exit status %d' % (skipifName, status))
```

Comment:
The start_test to JUnit script has different logic, and the attempts I made up to now were unsuccessful. The best I could do up to now was to get the failed skipif to show up but then the following test disappeared- the JUnit script ate that part of the log because the failed skipif did not emit exactly the right stuff.